### PR TITLE
more bugs fixes

### PR DESCRIPTION
**Pull request by kinarashah**
https://github.com/rancher/rancher/issues/25999: nodes get upgraded even if drain fails, this happens because node was considered prepared if node is cordoned/drained. If drain is set to true in cluster spec, only that should be considered. see https://github.com/rancher/rancher/pull/26016/files#diff-2eabb733bb0dba004b0c4c6d065eeb8cL170

https://github.com/rancher/rancher/issues/25946: even if unavailable nodes >= maxUnavailable nodes, we should still mark the upgraded nodes as active. there's no point in letting them stuck in upgrading. see https://github.com/rancher/rancher/pull/26016/files#diff-a16c43ed026936b620e2fdb9ca6196b8L273

https://github.com/rancher/rancher/issues/26023: all the fields should be read from cluster status, not spec. I am correctly doing this for rkeConfig, but maxUnavailable is getting read from spec. This ensures that we work with rkeConfig that has already gone through rke up. Fix is changing that`cluster.Spec` to `cluster.Status`.  

https://github.com/rancher/rancher/issues/25885: Changing log level for create/update node plan to debug.

### CODE REVIEW

**Comment by StrongMonkey** [comment]
Can we separate into two PRs? Or are those commits related?

**Comment by kinarashah** [comment]
@StrongMonkey I've put the first commit to a separate PR, and added explanation for each issue in the description

File: pkg/controllers/management/rkeworkerupgrader/upgrade.go [hunk 1]
```diff
@@ -234,16 +237,31 @@ func (uh *upgradeHandler) upgradeCluster(cluster *v3.Cluster, nodeName string) e
 		return err
 	}
 
-	toPrepareMap, toProcessMap, upgradedMap, notReadyMap, filtered, upgrading, done := uh.filterNodes(nodes, cluster.Status.NodeVersion)
+	upgradeStrategy := cluster.Status.AppliedSpec.RancherKubernetesEngineConfig.UpgradeStrategy
```
Comment (kinarashah) [inline pkg/controllers/management/rkeworkerupgrader/upgrade.go]: thinking if it'd make sense to short circuit if cluster condition is updating

File: pkg/controllers/management/rkeworkerupgrader/node.go [hunk 2]
```diff
@@ -162,12 +162,21 @@ func (uh *upgradeHandler) filterNodes(nodes []*v3.Node, expectedVersion int) (ma
...
+	return node.Spec.DesiredNodeUnschedulable == "true"
 }
 
-func preparedNode(node *v3.Node) bool {
-	return v3.NodeConditionDrained.IsTrue(node) || node.Spec.InternalNodeSpec.Unschedulable || v3.NodeConditionUpgraded.IsUnknown(node)
+func preparedNode(node *v3.Node, drain bool) bool {
+	if v3.NodeConditionUpgraded.IsUnknown(node) {
+		return true
+	}
+	if drain {
```
Comment (mrajashree) [inline pkg/controllers/management/rkeworkerupgrader/node.go]: should we do this check first before `if v3.NodeConditionUpgraded.IsUnknown(node)`?

**Pull request merged**